src/validation/rules/FieldsOnCorrectType.js

```diff
@@ -11,14 +11,70 @@
 import type { ValidationContext } from '../index';
 import { GraphQLError } from '../../error';
 import type { Field } from '../../language/ast';
-import type { GraphQLType } from '../../type/definition';
-
+import {
+  isAbstractType,
+  GraphQLAbstractType,
+  GraphQLObjectType,
+} from '../../type/definition';
 
 export function undefinedFieldMessage(
-  fieldName: string,
-  type: GraphQLType
-): string {
-  return `Cannot query field "${fieldName}" on "${type}".`;
+    field: string,
```

Comment:
I suggest keeping `fieldName` instead of `field` since it's slightly more clear that you're dealing with the name of something rather than the thing itself (could be misinterpreted as a field AST or field definition). I suggest actually renaming type to `typeName` since the type is now string